gymnasium/wrappers/record_episode_statistics.py

```diff
@@ -114,38 +85,30 @@ def step(self, action):
         ), f"`info` dtype is {type(infos)} while supported dtype is `dict`. This may be due to usage of other wrappers in the wrong order."
         self.episode_returns += rewards
         self.episode_lengths += 1
-        if not self.is_vector_env:
-            terminateds = [terminateds]
-            truncateds = [truncateds]
-        terminateds = list(terminateds)
-        truncateds = list(truncateds)
-
-        for i in range(len(terminateds)):
-            if terminateds[i] or truncateds[i]:
-                episode_return = self.episode_returns[i]
-                episode_length = self.episode_lengths[i]
-                episode_info = {
-                    "episode": {
-                        "r": episode_return,
-                        "l": episode_length,
-                        "t": round(time.perf_counter() - self.t0, 6),
-                    }
-                }
-                if self.is_vector_env:
-                    infos = add_vector_episode_statistics(
-                        infos, episode_info["episode"], self.num_envs, i
-                    )
-                else:
-                    infos = {**infos, **episode_info}
-                self.return_queue.append(episode_return)
-                self.length_queue.append(episode_length)
-                self.episode_count += 1
-                self.episode_returns[i] = 0
-                self.episode_lengths[i] = 0
+        dones = truncateds | terminateds
+        num_dones = np.sum(dones)
+        if num_dones:
+            episode_info = {
+                "episode": {
+                    "r": np.where(dones, self.episode_returns, 0),
+                    "l": np.where(dones, self.episode_lengths, 0),
+                    "t": np.full(
+                        (self.num_envs,), round(time.perf_counter() - self.t0, 6)
+                    ),
+                },
+            }
+            if self.is_vector_env:
+                episode_info["_episode"] = np.where(dones, True, False)
+            infos = {**infos, **episode_info}
```

Comment:
Could we change this to
```python
if "episode" or "_episode" in infos:
    # raise warning
else:
    infos["episode"] = {...}
    if self.is_vector_env:
        infos["_episodes"] = ...
```
This seems better than the old way as it is a single element we are adding to the dictionary